python/lsst/dm/rucio/register/rucio_interface.py

```diff
@@ -138,7 +138,44 @@ def _add_replicas(self, bundles: list[ResourceBundle]) -> None:
             A list of ResourceBundles
         """
         dids = [bundle.get_did() for bundle in bundles]
-        self.replica_client.add_replicas(rse=self.rse, files=dids)
+        retries = 0
+        max_retries = 5
+        while True:
+            try:
+                self.replica_client.add_replicas(rse=self.rse, files=dids)
+                break
+            except rucio.common.exception.RucioException:
+                retries += 1
+                if retries < max_retries:
+                    seconds = random.randint(10, 20)
+                    logger.debug(f"failed to add_replicas; sleeping {seconds} seconds")
+                    time.sleep(seconds)
+                    self.replica_client = ReplicaClient()  # XXX not sure we need to do this.
```

Comment:
You definitely do not need to remake the client.  The problem is on the server side, and the client does not have any relevant state.